### PR TITLE
Fixes for extended and logical

### PR DESCRIPTION
**Pull request by dbungert**
* handle the extended/logical flags
* make info available on storage/v2 API
* fix numbering of logical partitions

### CODE REVIEW

File: subiquity/server/controllers/filesystem.py [hunk 1]
```diff
@@ -375,7 +376,12 @@ async def v2_add_partition_POST(self, data: AddPartitionV2) \
             'mount': data.partition.mount,
         }
 
-        self.create_partition(disk, data.gap, spec, wipe='superblock')
+        flag = ""
+        if data.partition.partition_type in ('extended', 'logical'):
+            flag = data.partition.partition_type
+
```
Comment (mwhudson) [inline subiquity/server/controllers/filesystem.py:382]: Something somewhere should yell if you try to add an extended partition to a gpt disk I think.

Also, on the server side a Gap knows if it's in an extended partition, so in a sense sending partition_type == logical is redundant. Maybe we should extend the Gap that goes across the API to have that and then just have a boolean field to indicate creating an extended partition?

**Review by mwhudson — COMMENTED** [review]
Lots of questions I'm afraid!

**Comment by dbungert** [comment]
Replacing this with a newer PR.

File: subiquity/models/filesystem.py [hunk 2]
```diff
@@ -690,6 +690,10 @@ def serialize_number(self):
     def _number(self):
         if self.preserve:
             return self.number
+        elif self.flag == 'logical':
```
[the remark this answers is not in the document]
Comment (dbungert) [reply subiquity/models/filesystem.py:693]: #1354

**Pull request closed without merging**